### DISCUSSION TITLE
Is z-normalization of time series subsequences a good idea?

### DISCUSSION THREAD

**Discussion by seanlaw** [TDAmeritrade/stumpy#909] [Q&A]
Often, time series are de-trended or pre-processed before any analysis. With matrix profiles, this is overcome by z-normalizing each subsequence before computing the distance with respect to another subsequence. The question is whether or not z-normalization is a good idea?

**Comment by seanlaw** [comment]
According to Eammon Keogh:

> Z-normalization means we are ignoring the mean and standard deviation of the data. At least 99.9% of the time, that is the right thing to do.

Furthermore, this [2023 research paper](https://www.sciencedirect.com/science/article/abs/pii/S2214579623000400?via%3Dihub) by Felipe Tomazelli Lima and Vinicius M.A. Souza also seems to support this claim with the caveat that:

> alternative methods can outperform z-normalization in the classification task

**Comment by JaKasb** [comment]
Search for similar system-behavior, (sequences of up&down sections) : z-normalization
Search for similar magnitude or energy  : no-normalization

As alternative to mean-normalization : for preprocessing the time-series apply a high-pass fiter for removing slow drifts and varying bias-values.
E.g. window-size = 1 second -> lowest frequency in pattern = 1Hz -> Set high pass filter frequency below 1Hz to remove drift.

**Reply by NimaSarajpoor** [reply]
> Search for similar system-behavior, (sequences of up&down sections) : z-normalization
> 
> Search for similar magnitude or energy  : no-normalization

@JaKasb 
Correct. This is similar to what Eamonn said before. You cannot have both shape and magnitude. When shape matters, the magnitude does not, and vice versa.
